### ISSUE TITLE
SearchReadGroupSetsRequest "name" field ignored by searchReadGroupSets?

### ISSUE THREAD

**Issue by hjellinek** [ga4gh/server#534]
I'm using data from the server unit tests as the server's data.  The files are arranged like so:
```
dataset1/
    reads/
        1kg-low-coverage/
            ... BAM files ...
        wgBam/
            ... BAM files ...
    variants/
```

**Assumption**: each subdirectory in `reads` corresponds to a read group set.

Per the `SearchReadGroupSetsRequest` docs: "_Only return read group sets with this name (case-sensitive, exact match)._"  I create a search request with `datasetId` = `dataset1`; `name` = `"wgBam"`.

**Expected**:
The call to `searchReadGroupSets` will return a `SearchReadGroupSetsResponse` with its `readGroupSets` field containing a single read group set corresponding to `wgBam`.

**Actual**:
The call to `searchReadGroupSets` returns a `SearchReadGroupSetsResponse` with its `readGroupSets` field containing two read group sets, one apparently for `1kg-low-coverage` and one for `wgBam`.

**Comment by jeromekelleher** [comment]
Thanks for this @hjellinek, we missed this one. @dcolligan, looks we'll need something a bit more complicated for the ``readGroupSetsGenerator`` (paging is so tedious...).

**Comment by jeromekelleher** [comment]
An option here if this is too complicated to hit by the milestone is to raise a ``NotImplemented`  if someone specifies the name.

**Comment by macieksmuga** [comment]
@jeromekelleher, @dcolligan, @diekhans:  after discussing this with @adamnovak, @hjellinek and @hannes-ucsc, I see three, maybe four choices:

1. **Mark problem areas with `NotImplemented`** as you suggest above and move on. I'm very reluctant to adopt this solution, though, as it just kicks the can down the road. The overall purpose of baseline, after all, was to arrive at a set of schemas that we **do** have a full implementation of.
2. **Revise the baseline schedule** to give us time to deal properly with these issues. This is really a non-starter, considering that the baselining task has become a bottleneck for all other development, but I needed to list it anyway.
3. (no I'm not about to make a joke) **Remove paging from the schema methods**. Some form of paging could still be implemented by servers & clients, of course, as could streaming or some other mix of technical solutions to ensure the reliable transmission of large datasets. I'm just not convinced that it should be embedded into the schemas proper, where it gets in the way of the reference implementation (and presumably other implementations, that may wish to take a different approach to resumable transmission).

4. (suggested by @adamnovak) **Retain paging in the schemas as a stop-gap, but re-implement it to be based on byte ranges on the reference server** With the thought of eventually moving paging off the schemas proper, but keeping it there now to prevent further controversy. Here, instead of full-record based paging, let the page token be a start/end (or start/length, I don't care) byte range. The server performs all the back-end operations to fulfill the whole request (thus avoiding the pagination logic issues that are blocking other schema compliance work) but only sends the byte range requested. In this scheme, the server also doesn't need to send out a "next page token", which was a cause of further implementation issues if my memory serves me. Also, we can finally be rid of the maximum response byte size parameter, as it would be equivalent to specifying the paging token to that limit.

I would very much appreciate your thoughts on this, as I feel this is an urgent issue that could significantly affect the baseline project schedule & scope if not resolved ASAP.

**Comment by hjellinek** [comment]
From conversations I've overheard or read in my short time here, it seems like paging, at least as currently designed, is interfering with the design and implementation of certain basic operations.

Add to that the ongoing discussions of streaming as a replacement for paging, and it seems like the current paging regime may not be with us too much longer.  If so, then this might be a good time to pull it out of the schemas and ref server implementation.

(The byte-range idea is intriguing: HTTP supports it right out of the box, after all.  I can see some possible problems with it, which I'd be happy to discuss elsewhere, but I think it shares those with the existing paging scheme.)

**Comment by jeromekelleher** [comment]
I don't think this is that hard to fix @macieksmuga, it's just a bit tedious. Any major surgery to the paging protocol would be far more work than just fixing this particular problem. If not, then marking this with a `NotImplemented` is fine I think --- there are several other parts that are `NotImplemented`, and probably never will be implemented (i.e., searching variants by name, since this can't be done with htslib).

**Comment by macieksmuga** [comment]
In this case, @jeromekelleher, shouldn't we just remove said endpoint from the schema? If even the reference server never implements it...

**Comment by jeromekelleher** [comment]
It should be possible to implement this with a simple filter --- I'll have a go when I get a chance and some other PRs start landing. It won't be efficient for large numbers of readGroupSets, but I don't think it matters.

**Comment by dcolligan** [comment]
Fixed in #564
